dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -0,0 +1,119 @@
+﻿#include "pch.h"
+
+#include <NotificationsLongRunningProcess_h.h>
+
+#include "platform.h"
+#include "platformfactory.h"
+
+wil::unique_event s_event(wil::EventOptions::None);
+wil::unique_threadpool_timer s_timer;
+
+void NotificationsLongRunningPlatformImpl::Initialize()
+{
+    auto lock = m_lock.lock_exclusive();
+    THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
+
+    if (m_initialized)
+    {
+        return;
+    }
+
+    // Schedule event signaling after 30 seconds. This is in case we don't have any apps to track in the LRP.
+    // If we realize that we need to persist the LRP, timer should be canceled.
+    SetupTimer();
+
+    /* Verify registry and UDK list and make sure we have apps to be tracked */
+
+    /* Load your components */
+
+    m_initialized = true;
+}
+
+void NotificationsLongRunningPlatformImpl::Shutdown()
+{
+    auto lock = m_lock.lock_exclusive();
+    if (m_shutdown)
+    {
+        return;
+    }
+
+    /* Shut down your components */
+
+    m_shutdown = true;
+}
+
+void NotificationsLongRunningPlatformImpl::SignalEvent()
+{
+    s_event.SetEvent();
+}
+
+void NotificationsLongRunningPlatformImpl::WaitForEvent()
+{
+    s_event.wait();
```

Comment:
m_event. This isn't static right?

Reply:
I think there is a way to make the statics to be members. Let me take a look.

Reply:
why does this need to be static? Isnt it straightforward to just make this a member variable?

Reply:
CreateThreadpoolTimer doesn't have access to `this` object right now, so I made SignalEvent static. I think this can be moved to member if the PVOID on CreateThreadpoolTimer is `this`.